### PR TITLE
Enable CSI snapshot and clone feature

### PR DESCRIPTION
**Pull request by dsupure**
Enable CSI snapshot and clone features for the coming CSI 1.0.1 release.

### CODE REVIEW

**Comment by sdodsley** [comment]
I see you are using an rc1 image. Shouldnt we wait until the GA version is ready

**Comment by dsupure** [comment]
> I see you are using an rc1 image. Shouldnt we wait until the GA version is ready

Looks like they have 1.4 GA. Let me do a quick test.
https://quay.io/repository/k8scsi/csi-provisioner?tag=latest&tab=tags

**Comment by dsupure** [comment]
> > I see you are using an rc1 image. Shouldnt we wait until the GA version is ready
> 
> Looks like they have 1.4 GA. Let me do a quick test.
> https://quay.io/repository/k8scsi/csi-provisioner?tag=latest&tab=tags

I've verified 1.4 version works. Replace 1.4-rc1 to 1.4.

File: pure-csi/templates/provisioner.yaml [hunk 1]
```diff
@@ -78,7 +78,17 @@ spec:
...
               mountPath: /csi
-
+        - name: csi-snapshotter
+          {{- with .Values.csi.snapshotter.image }}
+          image: "{{ .name }}:v1.1.0"
+          {{- end }}
+          args:
+            - "--csi-address=/csi/csi.sock"
+            - "--connection-timeout=15s"
+            - "--leader-election=false"
```
Comment (trevormeiss) [inline pure-csi/templates/provisioner.yaml]: [question] why don't we use leader election?

File: pure-csi/templates/rbac.yaml [hunk 2]
```diff
@@ -44,16 +44,25 @@ rules:
...
     verbs: ["get", "list", "watch"]
   - apiGroups: [""]
     resources: ["nodes"]
     verbs: ["get", "list", "watch"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
+    resources: ["volumesnapshotclasses"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["apiextensions.k8s.io"]
+    resources: ["customresourcedefinitions"]
+    verbs: ["create", "list", "watch", "delete", "get", "update"]
```
Comment (trevormeiss) [inline pure-csi/templates/rbac.yaml]: why do we need these permissions for CRDs?

File: pure-csi/templates/provisioner.yaml (hunk 1 above)
Comment (dsupure) [reply pure-csi/templates/provisioner.yaml]: The setting is suggested by the external-snapshotter yaml [example](https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/setup-csi-snapshotter.yaml).

File: pure-csi/templates/rbac.yaml (hunk 2 above)
Comment (dsupure) [reply pure-csi/templates/rbac.yaml]: Same suggestion from the external-snapshotter deployment rbac [example](https://github.com/kubernetes-csi/external-snapshotter/blob/c933d6dacee9f7dc224c701c7579643c8ac700a5/deploy/kubernetes/rbac.yaml).

Comment (trevormeiss) [reply pure-csi/templates/rbac.yaml]: I guess CSI needs to create VolumeSnapshotClass, VolumeSnapshot, and VolumeSnapshotContent crds:
https://github.com/kubernetes-csi/external-snapshotter/blob/master/cmd/csi-snapshotter/create_crd.go

Comment (dsupure) [reply pure-csi/templates/rbac.yaml]: Yeah, you are right. Thanks @trevormeiss !

**Pull request merged**